modules/tensor_mechanics/include/materials/RadialReturnStressUpdate.h

```diff
@@ -28,26 +28,35 @@ InputParameters validParams<RadialReturnStressUpdate>();
  * Petrinic's Introduction to Computational Plasticity (2004) Oxford University Press.
  */
 
-class RadialReturnStressUpdate : public StressUpdateBase
+class RadialReturnStressUpdate : public StateUpdateBase
 {
 public:
   RadialReturnStressUpdate(const InputParameters & parameters);
 
   /// A radial return (J2) mapping method is defined in this material by overwritting
   /// the computeInelasticStrainIncrement method.
-  virtual void updateStress(RankTwoTensor & strain_increment,
-                            RankTwoTensor & inelastic_strain_increment,
-                            RankTwoTensor & stress_new) override;
+  virtual void updateState(RankTwoTensor & strain_increment,
+                           RankTwoTensor & inelastic_strain_increment,
+                           const RankTwoTensor & rotation_increment,
+                           RankTwoTensor & stress_new,
+                           const RankTwoTensor & stress_old,
+                           const RankFourTensor & elasticity_tensor,
+                           const RankTwoTensor & elastic_strain_old,
+                           bool compute_full_tangent_operator,
+                           RankFourTensor & tangent_operator) override;
```

Comment:
I don't think any of the tests are slower now.  That was a mistake i made with setting perform_finite_strain_rotations=true when i didn't need to.  But it's an interesting observations to keep in mind: when computing the full consistent tangent operator and using finite strains, _Jacobian_mult should be rotated.  This rotation is very computationally intensive.

So the extended interface to updateState has no affect on speed at all.  (And many of the additions to the interface were necessary because of the base_name thing and nothing else.)